text/0000-multi-part-examples-in-rustdoc/index.md

```diff
@@ -0,0 +1,231 @@
+- Feature Name: Multi-part examples in rustdoc
+- Start Date: 2021-02-09
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Support for multi-part examples in rustdoc, which consists of:
+  1. Running them as doc-tests
+  2. UI indicating members of multi-part examples in rendered docs
+
+# Motivation
+[motivation]: #motivation
+
+A long example often calls for lines of explanatory text interleaved within it.
+
+Currently, there are three means of achieving this:
+
+1. Use rust comments inside the long example.
+  Downsides:
+    1. No markdown
+    1. If a comment overflows the rendered code block horizontally, scrolling is possible, yet cumbersome, especially when the comment is far from the bottom, where the scroll bar happens to be.
+1. Split long examples into multiple code blocks and add the text between them.
+  Downsides:
+    1. Must disable doc-test-ing the example
+    1. Lack of UI indicating that the code blocks are multiple parts of a whole
+1. [The current documented workaround][current-documented-workaround]
```

Comment:
RFCs should stand on their own as much as possible, can you give a brief summary of what the workaround is? It looks like you mean adding `# ` to lines you don't want to show, which *can* be applied in this case, but I wouldn't describe as a workaround specifically for this problem.

Reply:
Better with c8cad888a81b3656561990a57f8d4242c1c3c2ed ?

Reply:
Yes, thanks.